src/store/conversationsStore.js

```diff
@@ -439,6 +440,20 @@ const actions = {
 		commit('addConversation', conversation)
 	},
 
+	async markConversationUnread({ dispatch, getters }, { token }) {
+		const conversation = Object.assign({}, getters.conversations[token])
+		if (!conversation) {
+			return
+		}
+
+		try {
+			await setConversationUnread(token)
+			dispatch('fetchConversation', { token })
+		} catch (error) {
+			console.debug('Error while marking conversation as unread: ', error)
+		}
```

Comment:
**question:** no `await dispatch('fetchConversation')` 

Is it intended that dispatch is called without await here?

Reply:
Nope, it's my mistake

Reply:
> I would say it's okay, as refreshing the conversation after doing the actual action should not be blocking?

`await` here has no affect on UI. The marker on UI updates when store update as the result of `fetchConversation` action. 

But without `await` this action resolves before the state is asctually updated and an exception from `fetchConversation` is not cough by this `catch`, making `unhandledrejection`.